addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -80,11 +80,10 @@ const Link = Widget.extend({
                 }
                 $node = $node.parent();
             }
-            const textLink = this.$link[0] || this.data.range.cloneContents();
-            const [encodedText, images] = this._encodeNodeToText(textLink);
-            this.data.originalText = this._decodeText(encodedText, images);
-            this.data.content = encodedText;
-            this.data.images = images;
+            const linkNode = this.$link[0] || this.data.range.cloneContents();
+            const linkText = linkNode.innerHTML || linkNode.textContent;
+            this.data.content = linkText.replace(/[ \t\r\n]+/g, ' ');
+            this.data.originalText = this.data.content;
```

Comment:
- Grepping this, I find the comment `// .html() is needed in case we selected images.`. Maybe worth updating?
- Had a **really** quick look but I don't think keeping that `originalText` is necessary, can't we use `.data.content` at all places?

Reply:
I updated the comment. About `originalText`, it is useful if and only if there is a place somewhere in Odoo where the user can separately set the `label` of the link like he used to in 14.0. I think this is not possible right now in 15.0, but do you think it's very wise to somewhat weaken that check ? I think you're right that it's not useful right now, but if we ever add back `label` somewhere else then we risk introducing new bugs.

Reply:
Ok, let's leave refactorings to master where they belong :+1: